tensorflow_graphics/rendering/camera/perspective.py

```diff
@@ -201,40 +201,42 @@ def intrinsics_from_matrix(matrix, name=None):
       "perspective_intrinsics_from_matrix".
 
   Returns:
-    Tuple of two tensors, each one of shape `[A1, ..., An, 2]`. The first
-    tensor represents the focal length, and the second one the principle point.
+    Tuple of three tensors, the first two of shape `[A1, ..., An, 2]` and
+    the third of shape `[A1, ..., An, 1]`. The first tensor represents the
+    focal length, and the second one the principle point and the third one
+    represents the skew coefficient.
 
   Raises:
     ValueError: If the shape of `matrix` is not supported.
   """
-  with tf.compat.v1.name_scope(name, "perspective_intrinsics_from_matrix",
-                               [matrix]):
+  with tf.compat.v1.name_scope(name, "perspective_intrinsics_from_matrix", [matrix]):
     matrix = tf.convert_to_tensor(value=matrix)
 
     shape.check_static(
         tensor=matrix,
         tensor_name="matrix",
         has_rank_greater_than=1,
         has_dim_equals=((-1, 3), (-2, 3)))
-
     fx = matrix[..., 0, 0]
     fy = matrix[..., 1, 1]
     cx = matrix[..., 0, 2]
     cy = matrix[..., 1, 2]
+    skew = matrix[..., 0, 1]
     focal = tf.stack((fx, fy), axis=-1)
     principal_point = tf.stack((cx, cy), axis=-1)
-  return focal, principal_point
+    skew = tf.expand_dims(skew, axis=0)
```

Comment:
Hey @ghosalsattam @fcole 

I hope you all had a great weekend

Sorry I didn't catch this earlier. But wouldn't this line create a tensor of shape `[1, A1, ... An]` instead of the desired `[A1, ... An, 1]`?

Shouldn't it be `skew = tf.expand_dims(skew, axis=-1)` instead?

Reply:
@fulkast, Thanks for pointing out and sorry for delayed response.
I have corrected this part.